lib/db.js

```diff
@@ -92,24 +92,45 @@ class LimitDBRedis extends EventEmitter {
     this.buckets[key] = utils.buildBucket(bucket);
   }
 
-  bucketKeyConfig(type, key) {
-    const fromCache = type.overridesCache.get(key);
+  paramsToBucketConfig(params) {
+    const bucketConfig = {};
+
+    if (params && params.configOverride) {
+      if (typeof params.configOverride.bucketSize === 'number') {
+        bucketConfig.size = params.configOverride.bucketSize;
+      }
+      if (typeof params.configOverride.refillRateMsPerToken === 'number') {
+        bucketConfig.per_interval = 1;
+        bucketConfig.interval = params.configOverride.refillRateMsPerToken;
+      }
+    }
```

Comment:
Would it make sense to follow the same input "shape" as the config object? What I mean is that the input config override should follow the same naming/shape of whatever is supported in the `buckets` param passed in the constructor (with attributes `size`, `per_second`, `per_hour`, `unlimited`). Although, note that the `overrides` attribute would not make sense in this case.

---

A problem with the approach here ☝️ is that the `bucketConfig.ttl` is not set properly. The `ttl` is necessary to automatically expire redis entries after their last use. The value is derived from the `per_interval` and `size` values as we can see here: https://github.com/auth0/limitd-redis/blob/2b44821a2f6bc0ad80a778310367483a0d2f7d9f/lib/utils.js#L32-L34

If we don't set it, we fallback to the `globalTTL` of 7 days, which means we retain redis entries longer than necessary.

--- 

The solution to this issue and what I mention above would probably be to re-use part of the `normalizeType()` or `buildBucket()` method from utils.js and pass the `configOverride` value to it. That way we know that whatever is passed as `configOverride` should be normalized the same way than the static config. Note that something is different in our case: the `bucketConfig.overrides` and `bucketConfig.overridesCache` should not be initialized when passing the `configOverride`.

Reply:
> Would it make sense to follow the same input "shape" as the config object? 

Can do it either way - whichever makes the most sense 🤷‍♂️ .

To me, I think this representation:
```
{
  size: 5,
  per_hour: 5,
}
```
Is tantamount to representing the Date with something like: `Aug 03, 2021`.  While it *can* be done (and often is, a la ISO 8601), a simple UNIX timestamp is often preferred.  I would argue that since we're dealing with a raw number (rate at which tokens should be restored to the bucket), that representing with something like a size + string/number indicator is silly.

So this was an attempt to kill the interval coming in as a strange data type.  But perhaps consistency needs to win out here.  Again, can do either 🤷‍♂️

Reply:
> A problem with the approach here ☝️ is that the bucketConfig.ttl is not set properly.

GREAT catch 👏 .  On it.

Reply:
> So this was an attempt to kill the interval coming in as a strange data type. But perhaps consistency needs to win out here.

Not arguing that the current format is the clearest. It was actually constrained by the old limitd format (we wanted to keep it as close as possible when building limitd-redis to ease the migration). I think consistency is important here (limitd-redis exposing a consistent interface across its API) even if the format is sub-par.

Reply:
Okie dokie, can intake that way too.  Will update.